upcase_booklist_app/upcase_booklist_app.py

```diff
@@ -1,8 +1,33 @@
-from flask import Flask 
+from flask import Flask, render_template, url_for
+from data import books
+
 app = Flask(__name__)
 
 
-@app.route('/')
-@app.route('/index')
+@app.route("/")
+@app.route("/index")
 def index():
-    return "Booklist App"
\ No newline at end of file
+    """Render the homepage"""
+    return render_template("index.html")
+
+
+@app.route("/books")
+def book_list():
+    """Show the list of books"""
+    return render_template("books.html", books=books)
+
+
+@app.route("/books/<book_id>")
+def single_book(book_id):
+    """Show details for a specific book."""
+    return render_template("single_book.html", book=books[book_id])
+
+
+@app.route("/authors")
+def authors():
+    """Show the list of authors and the books they've written"""
+    return render_template("authors.html", books=books)
```

Comment:
Consider creating a new data file for `authors` and feeding this up as well.

Reply:
@paulghaddad I'm not sure why I'd do this? All of the data that's in books I need for the authors view function. So it'd be the exact same data file, and that seems redundant -- right?